src/streaming.rs

```diff
@@ -0,0 +1,270 @@
+use ::open_vcdiff_sys;
+
+use std::error;
+use std::fmt;
+use std::marker::PhantomData;
+use std::mem;
+use std::os::raw::c_void;
+use std::slice;
+use std::result;
+
+#[derive(Debug,Copy,Clone,PartialEq,Eq)]
+pub struct Error(&'static str);
```

Comment:
Error handling in general is really unfortunate here. I'd love to capture the results from the library, but it [writes straight to `std::cerr`](https://github.com/google/open-vcdiff/blob/master/src/logging.h#L35-L45). I was thinking about redefining the macros so that it would write to a `stringstream`, but… reentrancy. Error capture would have to be thread local.

The `thread_local` storage class is C++11, and I didn't want to make any particular demands of the C++ compiler, and of course Rust [`thread_local!` would be readily available](https://doc.rust-lang.org/std/macro.thread_local.html) instead. But then… what, C++ calls Rust to get a pointer in which to store a `stringstream` where it can write errors, and then Rust calls C++ to ask what's in the `stringstream`? That doesn't seem right.

That's about where I got before settling for `Error(&'static str)` :-)